### PR TITLE
Merge pull request #1 from dpastoor/master

### PR DESCRIPTION
**Pull request by nmehta2**
let's try this again

### CODE REVIEW

File: lo_2010.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// from lo et al 2010 - Population pharmacokinetics of vancomycin in premature Malaysian neonates:
+// identification of predictors for dosing determination.
+
+[PARAM] @annotated
+CL  : 0.345 : Clearance (L/hr)
+V   : 1.75  : Volume (L) 
+wt  : 2.9   : Weight (kg)
+pma : 34.8  : post-menstral age (wk)
+scr : 0.9   : serum creatinine (mg/dL)
+Fsga :   ?  : small-for-gestational-age
```
Comment (dpastoor) [inline lo_2010.cpp]: Set default of 1

File: lo_2010.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+spiro :   ? : spironolactone (L/kg)
```
Comment (dpastoor) [inline lo_2010.cpp]: set default of 0

File: lo_2010.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+CENT : Central compartment (mg)
+
+[PKMODEL]
+ncmt=1, trans=11
+
+[MAIN]
+D_CENT = 1; 
+double NORM_WT = 70;
+double NORM_PMA = 30;
+double Fsga = ?;
```
Comment (dpastoor) [inline lo_2010.cpp]: this is how you define values inside your code - you have already set these as parameters so you don't need to re-define them

**Review by dpastoor — CHANGES_REQUESTED** [review]

**Comment by dpastoor** [comment]
so keep in mind distinguishing between parameters for input, and things you define in `[MAIN]`

the syntax of `double <value> = ...` is you initializing a value to use later on. For exaple the normalization value for wt, etc. If you are already passing that value in as a parameter it is already available so you don't want to define.

**Comment by nmehta2** [comment]
Why can't you just use the number 70 instead of creating a parameter in `[MAIN]` for `NORM_WT`? Is it easier to have a parameter which you can change if the norm_wt value changes (ex. different patient demographic)?

**Comment by dpastoor** [comment]
good question - you are close - the biggest reason is that it makes it much clearer to read, especially as the number of parameters expands. For example

`pow(WT/70, 0.75)` doesn't necessarily convey why 70 was chosen and also starts to become harder to read when in a larger formula

whereas by defining the values above with names, it becomes clearer to read the equations, as well as update/change accordingly.

File: lo_2010.cpp [hunk 4]
```diff
@@ -22,8 +22,6 @@ ncmt=1, trans=11
 D_CENT = 1; 
 double NORM_WT = 70;
 double NORM_PMA = 30;
-double Fsga = ?;
-double spiro = ?;
 double CLi = 1.0 * pow((wt/NORM_WT), 0.75) * pow((pma/NORM_PMA), 3.16) * Fsga
```
Comment (dpastoor) [inline lo_2010.cpp]: ok so looking in the actual publication, this F term is misleading. I originally thought it was a 1/0, but that wouldn't makes sense as if the baby wasn't small for gestantioal age, CLi would be 0. Looking into the actual publication, we see that the full CL equation was actually

```
CL = 1.0 * (wt/70)^0.75*(PMA/30)^3.16*[0.83*SGA + 1.03*(1-SGA)]
```

so we should expand this equation to just have the parameter be called SGA (with a default of 0) and then use this equation.

**Review by dpastoor — CHANGES_REQUESTED** [review]

**Comment by nmehta2** [comment]
ah okay, that makes sense! Thank you for explaining!

File: models/andeson_2007.cpp [hunk 5]
```diff
@@ -23,8 +23,8 @@ ncmt=1, trans=11
 D_CENT = 1; 
 double NORM_WT = 70;
 double NORM_PMA = 34.8;
-double CLi = 3.83*pow(wt/NORM_WT,0.75)*(pow(pma, 3.68)/(pow(pma, 3.68) + pow(33.3, 3.68)))*(516*pow(e, (0.00823*((PMA-40)/52-40)))/scr)/6)*Fventilation;
-double Vi = 39.4*(wt/NORM_WT)*Finotrope;
+double CLi = 3.83*pow(wt/NORM_WT,0.75)*(pow(pma, 3.68)/(pow(pma, 3.68) + pow(33.3, 3.68)))*(516*pow(e, (0.00823*((PMA-40)/52-40)))/scr)/6)*Fvent;
```
Comment (dpastoor) [inline models/andeson_2007.cpp]: so these are both incorrect. Think of what you are setting as of now. You are saying multiply by 0, so both CL/V will be 0.

In reality, similar to the lo paper, these are likely placeholders for more complex equations like `(1.5*on_vent)`. The `Fvent` stands, essentially, for fraction corresponding to vent status. We need to actually go through the original anderson paper to find the actual values

File: models/andeson_2007.cpp [hunk 6]
```diff
@@ -23,8 +23,8 @@ ncmt=1, trans=11
...
+double Vi = 39.4*(wt/NORM_WT)*Finot;
 
 [OMEGA] @annotated
   ECL : 0.04 : Eta on CL
```
Comment (dpastoor) [inline models/andeson_2007.cpp]: we will also need to pull out the values for the random effects

**Review by dpastoor — CHANGES_REQUESTED** [review]
getting closer!

File: models/lo_2010.cpp [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+[MAIN]
+D_CENT = 1; 
+double NORM_WT = 70;
+double NORM_PMA = 30;
+double CLi = 1.0 * pow((wt/NORM_WT), 0.75) * pow((pma/NORM_PMA), 3.16) * (0.83*SGA + 1.03*(1-SGA))
+double Vi = 0.572 * (1-0.344*spiro) * wt
+
+[OMEGA] @annotated
+  ECL : 0.04 : Eta on CL
```
Comment (dpastoor) [inline models/lo_2010.cpp]: @nmehta2 so close! lets just get the lo omega and sigma values and i think you got it!

File: models/lo_2010.cpp [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  EV  : 0.01 : Eta on V
+  
+  [SIGMA] @annotated
+  PROP : 0.04 : Proportional error (%CV)
+  ADD  : 1.14  : Additive (SD)
+  
+  [TABLE]
+double CP = CENT/Vi;
+double DV = CP*(1+PROP);
```
Comment (dpastoor) [inline models/lo_2010.cpp]: check the residual error structure as well. This is only proportional as of now - is there also an additive component?

**Review by dpastoor — CHANGES_REQUESTED** [review]
so close!

File: models/lo_2010.cpp (hunk 7 above)
Comment (nmehta2) [reply models/lo_2010.cpp]: i updated the omega values and the residual error model values! are they correct?

File: models/lo_2010.cpp (hunk 8 above)
Comment (nmehta2) [reply models/lo_2010.cpp]: added the additive comp! did i write it correctly?

File: models/lo_2010.cpp [hunk 9]
```diff
@@ -26,12 +26,12 @@ double CLi = 1.0 * pow((wt/NORM_WT), 0.75) * pow((pma/NORM_PMA), 3.16) * (0.83*S
 double Vi = 0.572 * (1-0.344*spiro) * wt
 
 [OMEGA] @annotated
-  ECL : 0.04 : Eta on CL
-  EV  : 0.01 : Eta on V
+  ECL : 20.5 : Eta on CL
```
Comment (dpastoor) [inline models/lo_2010.cpp]: what are all these values? they still need to be reported as variances, so you were correct in that 0.04 --> 20%CV. Remember the variance to %CV calculation can be approximated by sqrt(variance)*100, so sqrt(0.04)*100 = 20%

**Pull request merged**